airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/operation/DefaultCheckOperation.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.core.operation
+
+import io.airbyte.cdk.core.context.env.ConnectorConfigurationPropertySource
+import io.airbyte.cdk.core.operation.executor.OperationExecutor
+import io.airbyte.protocol.models.v0.AirbyteConnectionStatus
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.micronaut.context.annotation.Requires
+import jakarta.inject.Named
+import jakarta.inject.Singleton
+
+private val logger = KotlinLogging.logger {}
+
+@Singleton
+@Named("checkOperation")
+@Requires(
+    property = ConnectorConfigurationPropertySource.CONNECTOR_OPERATION,
+    value = "check",
+)
+class DefaultCheckOperation(
+    @Named("checkOperationExecutor") private val operationExecutor: OperationExecutor,
+) : Operation {
```

Comment:
where is Operation defined? It looks like it is not helping us express the return values of the various execute methods

Reply:
There is an interface in the same package that defines the methods.